src/App.tsx

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useState } from 'react';
+import DailyIframe, { DailyCall, DailyBrowserInfo } from '@daily-co/daily-js';
+import { DailyProvider } from '@daily-co/daily-react';
+import { Link, Outlet, useLocation } from 'react-router-dom';
+
+import { logger } from './utils/Logger.ts';
+
+import { UnsupportedBrowser } from './components/UnsupportedBrowser';
+import { Navigation } from './components/Navigation';
+import './App.css';
+import { DailyLogo } from './components/shared/DailyLogo';
+
+export const App = () => {
+  const [daily, setDaily] = useState<DailyCall | null>(null);
+  const [browserInfo, setBrowserInfo] = useState<DailyBrowserInfo | null>(null);
+
+  useEffect(() => {
+    if (daily) return;
+    const supportedBrowser = DailyIframe.supportedBrowser();
+    setBrowserInfo(supportedBrowser);
+
+    if (supportedBrowser.supported) {
+      logger.info('Browser is supported, going to create the call object!');
+      const instance = DailyIframe.getCallInstance();
+      if (instance) {
+        logger.info(
+          'Browser is supported, but there is already a call object.',
+        );
+        setDaily(instance);
+      } else {
+        const co = DailyIframe.createCallObject();
+        setDaily(co);
+        logger.info('Created call object');
+      }
+    } else {
+      logger.warn('Browser is not supported. Not creating a call object.');
+    }
+  }, [daily]);
+
+  const location = useLocation();
+
+  useEffect(() => {
+    if (!daily || daily.isDestroyed()) return;
+
+    const startCam = async () => {
+      console.log('going to start cam..');
+      await daily?.startCamera();
+    };
+
+    startCam()
+      .then(() => logger.info('Camera started. Ready to start testing!'))
+      .catch((e) => logger.error(e));
+  }, [daily]);
+
+  return (
+    <div className="app">
+      <header>
+        <DailyLogo gradientId="dailyslashdesktop" height="32" />
+        <p>
+          The Daily API makes implementing a precall test a breeze, especially
+          when combined with the{' '}
+          <a href="https://docs.daily.co/reference/daily-react" target="_blank">
+            Daily React library
+          </a>
+          . This app demonstrates how to do network and device checks using{' '}
+          <a href="https://docs.daily.co/reference/rn-daily-js/instance-methods/test-connection-quality">
+            testConnectionQuality
+          </a>
+          ,{' '}
+          <a href="https://docs.daily.co/reference/daily-js/instance-methods/test-network-connectivity">
+            testNetworkConnectivity()
+          </a>
+          , and{' '}
+          <a href="https://docs.daily.co/reference/daily-js/instance-methods/test-websocket-connectivity">
+            testWebsocketConnectivity
+          </a>
```

Comment:
I believe you'd want these links to be external, too, right?
```suggestion
          <a href="https://docs.daily.co/reference/rn-daily-js/instance-methods/test-connection-quality" target="_blank" rel="noreferrer">
            testConnectionQuality
          </a>
          ,{' '}
          <a href="https://docs.daily.co/reference/daily-js/instance-methods/test-network-connectivity" target="_blank" rel="noreferrer">
            testNetworkConnectivity()
          </a>
          , and{' '}
          <a href="https://docs.daily.co/reference/daily-js/instance-methods/test-websocket-connectivity" target="_blank" rel="noreferrer">
            testWebsocketConnectivity
          </a>
```